library/src/main/java/no/finntech/capturandro/DownloadMultipleAsyncTask.java

```diff
@@ -15,66 +15,67 @@
-    private final ContentResolver contentResolver;
     private final CapturandroCallback.ImageHandler imageHandler;
     private final int longestSide;
     private final ClipData clipData;
     private static final Handler handler = new Handler(Looper.getMainLooper());
+    private final Context context;
 
-    public DownloadMultipleAsyncTask(ContentResolver contentResolver, CapturandroCallback.ImageHandler imageHandler, int longestSide, ClipData clipData) {
+    public DownloadMultipleAsyncTask(Context context, CapturandroCallback.ImageHandler imageHandler, int longestSide, ClipData clipData) {
         this.clipData = clipData;
-        this.contentResolver = contentResolver;
+        this.context = context;
         this.imageHandler = imageHandler;
         this.longestSide = longestSide;
     }
 
     @Override
     protected Void doInBackground(Void... params) {
-        List<String> filenames = new ArrayList<>();
+        List<UUID> importIds = new ArrayList<>();
+
         for (int i = 0; i < clipData.getItemCount(); i++) {
-            filenames.add(getUniqueFilename());
+            importIds.add(UUID.randomUUID());
         }
 
         // Notify that we've started X gallery imports
         for (int i = 0; i < clipData.getItemCount(); i++) {
-            final String filename = filenames.get(i);
+            final UUID importId = importIds.get(i);
             handler.post(new Runnable() {
                 @Override
                 public void run() {
-                    imageHandler.onGalleryImportStarted(filename);
+                    imageHandler.onGalleryImportStarted(importId);
                 }
             });
         }
 
         for (int i = 0; i < clipData.getItemCount(); i++) {
-            final String filename = filenames.get(i);
-            try {
-                Uri uri = clipData.getItemAt(i).getUri();
-                int orientation = GalleryHandler.getOrientation(contentResolver, uri);
-                Bitmap mediaStoreBitmap = MediaStore.Images.Media.getBitmap(contentResolver, uri);
-                final Bitmap bitmap = BitmapUtil.getProcessedBitmap(mediaStoreBitmap, longestSide, orientation);
-                handler.post(new Runnable() {
-                    @Override
-                    public void run() {
-                        imageHandler.onGalleryImportSuccess(filename, bitmap);
-                    }
-                });
-            } catch (final IOException e) {
-                e.printStackTrace();
-                handler.post(new Runnable() {
-                    @Override
-                    public void run() {
-                        imageHandler.onGalleryImportFailure(filename, e);
-                    }
-                });
-            }
+            final UUID importId = importIds.get(i);
+            Uri uri = clipData.getItemAt(i).getUri();
+            int orientation = GalleryHandler.getOrientation(context.getContentResolver(), uri);
```

Comment:
I notice we have a GalleryHandler.getOrientation and a getOrientation in BitmapUtil. I see they are different, but could we get the orientation without going through bitmaputil more often? This should be miles faster and probably far more memory efficient than decoding the bitmap.